src/generator/operations.ts

```diff
@@ -257,12 +270,26 @@ export async function generateOperations(session: Session<CodeModel>): Promise<O
         parameterizedURL = 'hostURL';
       }
     }
+    // populate any default values.  this only applies to non-ARM
+    // scenarios as they don't generate an options struct.
+    if (!isARM) {
+      for (const optionalParam of values(optionalParams)) {
+        if (optionalParam.clientDefaultValue) {
+          clientText += `\tif ${optionalParam.language.go!.name} == nil {\n`;
+          clientText += `\t\t${optionalParam.language.go!.name}Default := ${getClientDefaultValue(optionalParam)}\n`;
+          clientText += `\t\t${optionalParam.language.go!.name} = &${optionalParam.language.go!.name}Default\n`;
+          clientText += '\t}\n';
+        }
+      }
+    }
     // construct client literal
     clientText += `\tclient := &${clientName}{\n`;
     // populate any default values
-    for (const optionalParam of values(optionalParams)) {
-      if (optionalParam.clientDefaultValue) {
-        clientText += `\t\t${optionalParam.language.go!.name}: ${getClientDefaultValue(optionalParam)},\n`;
+    if (isARM) {
```

Comment:
Is there an output example for ARM?

Reply:
I can't think of one for Azure ARM, but for vanilla ARM this is used in some of the test server specs, see [this one](https://github.com/Azure/autorest.testserver/blob/main/swagger/custom-baseUrl.json) for an example (the `host` param).